tools/python/boutiques/localExec.py

```diff
@@ -80,16 +80,26 @@ def execute(self, mount_strings):
     # Note that docker/singularity cannot do a local volume mount of files starting with a '.', hence this one does not
     millitime = int(time.time()*1000)
     dsname = 'temp-' + str(random.SystemRandom().randint(0,int(millitime))) + "-" + str(millitime) + '.localExec.boshjob.sh' # time tag to avoid overwrites
+    dsname = op.realpath(dsname)
     # If container is present, alter the command template accordingly
     if conIsPresent:
       if conType == 'docker':
         # Pull the docker image
-        if self._localExecute("docker pull " + str(conImage)):
+        if self._localExecute("docker pull " + str(conImage))[1]:
           print("Container not found online - trying local copy")
       elif conType == 'singularity':
-        # Pull the docker image
-        if self._localExecute("singularity pull shub://" + str(conImage)):
-          print("Container not found online - trying local copy")
+        if not conIndex:
+          conIndex = "shub://"
+        elif not conIndex.endswith("://"):
+          conIndex = conIndex + "://"
+        conName = conImage.replace("/", "-") + ".simg"
+
+        if conName not in os.listdir('./'):
+          print(os.listdir('./'))
+          # Pull the singularity image
+          if self._localExecute("singularity pull --name \"{}\" {}{}".format(conName, conIndex, conImage))[1]:
```

Comment:
It would be nice if this could print the command that is being executed as bosh would typically hang on that for a while. Maybe add a print in localExecute?